pkg/webhook/multiclusterservice/validating.go

```diff
@@ -0,0 +1,129 @@
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	networkingv1alpha1 "github.com/karmada-io/karmada/pkg/apis/networking/v1alpha1"
+	"github.com/karmada-io/karmada/pkg/util/lifted"
+)
+
+// ValidatingAdmission validates MultiClusterService object when creating/updating.
+type ValidatingAdmission struct {
+	decoder *admission.Decoder
+}
+
+// Check if our ValidatingAdmission implements necessary interface
+var _ admission.Handler = &ValidatingAdmission{}
+var _ admission.DecoderInjector = &ValidatingAdmission{}
+
+// Handle implements admission.Handler interface.
+// It yields a response to an AdmissionRequest.
+func (v *ValidatingAdmission) Handle(ctx context.Context, req admission.Request) admission.Response {
+	mcs := &networkingv1alpha1.MultiClusterService{}
+	err := v.decoder.Decode(req, mcs)
+	if err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+	klog.Infof("Validating MultiClusterService(%s/%s) for request: %s", mcs.Namespace, mcs.Name, req.Operation)
+
+	if req.Operation == admissionv1.Update {
+		oldMcs := &networkingv1alpha1.MultiClusterService{}
+		err = v.decoder.DecodeRaw(req.OldObject, oldMcs)
+		if err != nil {
+			return admission.Errored(http.StatusBadRequest, err)
+		}
+		if errs := v.validateMCSUpdate(oldMcs, mcs); len(errs) != 0 {
+			klog.Errorf("%v", errs)
+			return admission.Denied(errs.ToAggregate().Error())
+		}
+	} else {
+		if errs := v.validateMCS(mcs); len(errs) != 0 {
+			klog.Errorf("%v", errs)
+			return admission.Denied(errs.ToAggregate().Error())
+		}
+	}
+	return admission.Allowed("")
+}
+
+// InjectDecoder implements admission.DecoderInjector interface.
+// A decoder will be automatically injected.
+func (v *ValidatingAdmission) InjectDecoder(d *admission.Decoder) error {
+	v.decoder = d
+	return nil
+}
+
+func (v *ValidatingAdmission) validateMCSUpdate(oldMcs, newMcs *networkingv1alpha1.MultiClusterService) field.ErrorList {
+	allErrs := apimachineryvalidation.ValidateObjectMetaUpdate(&newMcs.ObjectMeta, &oldMcs.ObjectMeta, field.NewPath("metadata"))
+	allErrs = append(allErrs, v.validateMCS(newMcs)...)
+	allErrs = append(allErrs, lifted.ValidateLoadBalancerStatus(&newMcs.Status.LoadBalancer, field.NewPath("status", "loadBalancer"))...)
+	return allErrs
+}
+
+func (v *ValidatingAdmission) validateMCS(mcs *networkingv1alpha1.MultiClusterService) field.ErrorList {
+	allErrs := apimachineryvalidation.ValidateObjectMeta(&mcs.ObjectMeta, true,
+		apimachineryvalidation.NameIsDNS1035Label, field.NewPath("metadata"))
+	allErrs = append(allErrs, v.validateMultiClusterServiceSpec(mcs)...)
+	return allErrs
+}
+
+// validateMultiClusterServiceSpec validates MultiClusterService spec.
+func (v *ValidatingAdmission) validateMultiClusterServiceSpec(mcs *networkingv1alpha1.MultiClusterService) field.ErrorList {
+	allErrs := field.ErrorList{}
+	specPath := field.NewPath("spec")
+	var allPortNames map[string]bool
```

Comment:
seems it's better to use `sets.Set`

Reply:
Thanks, it is more reasonable to use `sets.Set`.
Done.